ui/src/core/xibo-calendar.js

```diff
@@ -40,6 +40,133 @@ $(document).ready(function() {
         });
     });
 
+    // Set a listener for type change event
+    $('body').on('change', 'select[name="criteria_type[]"]', function() {
+        // Get the row where the type was changed
+        var $row = $(this).closest('.form-group');
```

Comment:
Instead of using this repeatedly in the event handler function, I would instead capture the event in the anonymous function:
```
function(e) {
(...)
```

And then create a simple variable from the target at the top of the function:
```
var $target = $(e.target);
```

To use instead of `$(this)`. This avoids any unwanted scope issues that might come with the word `this`.